clients/drcachesim/reader/reader.h

```diff
@@ -190,6 +190,9 @@ class reader_t : public std::iterator<std::input_iterator_tag, memref_t>,
     bool
     is_record_synthetic() const override
     {
+        if (cur_ref_.marker.type == TRACE_TYPE_MARKER &&
+            cur_ref_.marker.marker_type == TRACE_MARKER_TYPE_WAIT)
+            return true;
         return suppress_ref_count_ >= 0;
```

Comment:
I don't remember how this logically implies a synthetic record. Please add a comment.

Reply:
Done.  These are synthesized by the framework and not part of the original inputs nor counting toward ordinals.  See the docs for is_record_synthetic().